src/gz/mod.rs

```diff
@@ -316,11 +316,11 @@ mod tests {
             .extra(vec![0, 1, 2, 3])
             .read(&r[..], Compression::default());
         let mut d = read::GzDecoder::new(e);
-        assert_eq!(d.header().unwrap().filename(), Some(&b"foo.rs"[..]));
-        assert_eq!(d.header().unwrap().comment(), Some(&b"bar"[..]));
-        assert_eq!(d.header().unwrap().extra(), Some(&b"\x00\x01\x02\x03"[..]));
         let mut res = Vec::new();
         d.read_to_end(&mut res).unwrap();
+        assert_eq!(d.header().unwrap().filename(), Some(&b"foo.rs"[..])); // must read first
```

Comment:
Usually in async code, we don't want to read IO before poll. The new behavior does not allow us to get gzip header before read.

I don't know if it's worth release 2.0 for this.